internal/test/e2e/host.go

```diff
@@ -56,250 +52,126 @@ type stubDataMonitor struct{}
+	defer eps.mu.Unlock()
+	t, ok := eps.bans[addr]
+	return ok && time.Now().Before(t), nil
 }
 
-// ProcessConsensusChange implements the modules.ConsensusSetSubscriber interface.
-func (m *chainManager) ProcessConsensusChange(cc modules.ConsensusChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.tip = consensus.State{
-		Network: m.network,
-		Index: types.ChainIndex{
-			ID:     types.BlockID(cc.AppliedBlocks[len(cc.AppliedBlocks)-1].ID()),
-			Height: uint64(cc.BlockHeight),
-		},
+func newEphemeralPeerStore() syncer.PeerStore {
+	return &ephemeralPeerStore{
+		peers: make(map[string]syncer.PeerInfo),
+		bans:  make(map[string]time.Time),
 	}
-	m.synced = synced(cc.AppliedBlocks[len(cc.AppliedBlocks)-1].Timestamp)
 }
 
-// Network returns the network name.
-func (m *chainManager) Network() string {
-	switch m.network.Name {
-	case "zen":
-		return "Zen Testnet"
-	case "mainnet":
-		return "Mainnet"
-	default:
-		return m.network.Name
-	}
+type cmShim struct {
+	*chain.Manager
 }
 
-// Close closes the chain manager.
-func (m *chainManager) Close() error {
-	select {
-	case <-m.close:
-		return nil
-	default:
+func (c cmShim) IndexAtHeight(height uint64) (types.ChainIndex, error) {
+	index, ok := c.BestIndex(height)
+	if !ok {
+		return types.ChainIndex{}, fmt.Errorf("no index at height %d", height)
 	}
-	close(m.close)
-	return m.cs.Close()
+	return index, nil
 }
 
-// Synced returns true if the chain manager is synced with the consensus set.
-func (m *chainManager) Synced() bool {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	return m.synced
-}
-
-// BlockAtHeight returns the block at the given height.
-func (m *chainManager) BlockAtHeight(height uint64) (types.Block, bool) {
-	sb, ok := m.cs.BlockAtHeight(stypes.BlockHeight(height))
-	var c types.Block
-	utils.ConvertToCore(sb, (*types.V1Block)(&c))
-	return types.Block(c), ok
-}
-
-func (m *chainManager) LastBlockTime() time.Time {
-	return time.Unix(int64(m.cs.CurrentBlock().Timestamp), 0)
-}
-
-// IndexAtHeight return the chain index at the given height.
-func (m *chainManager) IndexAtHeight(height uint64) (types.ChainIndex, error) {
-	block, ok := m.cs.BlockAtHeight(stypes.BlockHeight(height))
+func (c cmShim) BlockAtHeight(height uint64) (types.Block, bool) {
+	index, ok := c.BestIndex(height)
 	if !ok {
-		return types.ChainIndex{}, ErrBlockNotFound
+		return types.Block{}, false
 	}
-	return types.ChainIndex{
-		ID:     types.BlockID(block.ID()),
-		Height: height,
-	}, nil
+	return c.Block(index.ID)
 }
 
-// TipState returns the current chain state.
-func (m *chainManager) TipState() consensus.State {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	return m.tip
-}
-
-// AcceptBlock adds b to the consensus set.
-func (m *chainManager) AcceptBlock(b types.Block) error {
-	var sb stypes.Block
-	utils.ConvertToSiad(types.V1Block(b), &sb)
-	return m.cs.AcceptBlock(sb)
+func (c cmShim) Subscribe(s modules.ConsensusSetSubscriber, ccID modules.ConsensusChangeID, cancel <-chan struct{}) error {
+	return nil
```

Comment:
Quite sure this is causing the tests to fail @lukechampine. Cause subscriptions won't work.

Reply:
@lukechampine if you upgrade our `hostd` dependency to https://github.com/SiaFoundation/hostd/pull/439 you should be able to wrap this PR up now. 
The `hostd` PR isn't ready for its `master` branch but since we only use it for testing it should be fine to upgrade.